### PR TITLE
Fix user mention bug when parsing markdown documents

### PR DESCRIPTION
**Pull request by deringer**
Adjust MentionPattern to not match users mid-sentence (or email addresses)

Fix link to user profile, update based on adjusted MentionPattern.

### CODE REVIEW

**Comment by deringer** [comment]
This reared it's head in my README.md as I had an email address in the document, which caused it to be rendered in anchor tags.

**Comment by deringer** [comment]
I've also just added in a little bit of padding on paragraph tags within the markdown styles.

File: modules/base/markdown.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func (options *CustomRender) Image(out *bytes.Buffer, link []byte, title []byte,
 }
 
 var (
-	MentionPattern     = regexp.MustCompile(`@[0-9a-zA-Z_]{1,}`)
+	MentionPattern     = regexp.MustCompile(`(\s@)[0-9a-zA-Z_]{1,}`)
```
Comment (tboerger) [inline modules/base/markdown.go]: And what happens if the text starts with a mention?

File: public/ng/css/gogs.css [hunk 2]
```diff
@@ -448,6 +448,9 @@ img.avatar-100 {
   margin: 15px 0;
   border-bottom: 2px solid #EEE;
 }
+.markdown p {
```
Comment (tboerger) [inline public/ng/css/gogs.css]: I think this should go into one of the less files?

Comment (Unknwon) [inline public/ng/css/gogs.css]: I believe so.

**Comment by jcracknell** [comment]
This would fix #737.

**Comment by Unknwon** [comment]
@jcracknell thanks your info!

File: modules/base/markdown.go (hunk 1 above)
Comment (jcracknell) [inline modules/base/markdown.go]: Probably want `(^|\s)[0-9a-zA-Z_]{1,}`.

File: modules/base/markdown.go [hunk 3]
```diff
@@ -123,7 +123,7 @@ func RenderSpecialLink(rawBytes []byte, urlPrefix string) []byte {
 			ms := MentionPattern.FindAll(line, -1)
 			for _, m := range ms {
 				line = bytes.Replace(line, m,
-					[]byte(fmt.Sprintf(`<a href="%s/user/%s">%s</a>`, setting.AppSubUrl, m[1:], m)), -1)
+					[]byte(fmt.Sprintf(`<a href="%s/%s">%s</a>`, setting.AppSubUrl, m[2:], m)), -1)
```
Comment (jcracknell) [inline modules/base/markdown.go]: Don't forget to put the space back.

Comment (Unknwon) [inline modules/base/markdown.go]: What space do you mean?

**Comment by Unknwon** [comment]
Merged, thanks again!

**Pull request merged**